llvm/lib/Transforms/Instrumentation/MemProfiler.cpp

```diff
@@ -824,11 +824,14 @@ memprof::extractCallsFromIR(Module &M) {
           continue;
 
         StringRef CalleeName = CalledFunction->getName();
-        uint64_t CallerGUID =
-            IndexedMemProfRecord::getGUID(DIL->getSubprogramLinkageName());
-        uint64_t CalleeGUID = IndexedMemProfRecord::getGUID(CalleeName);
-        LineLocation Loc = {GetOffset(DIL), DIL->getColumn()};
-        Calls[CallerGUID].emplace_back(Loc, CalleeGUID);
+        do {
+          uint64_t CallerGUID =
+              IndexedMemProfRecord::getGUID(DIL->getSubprogramLinkageName());
```

Comment:
Should we assert that SubProgramLinkageName is not empty? 
Same for the usage on L833.

Reply:
The module debug flags include `debugInfoForProfiling` which we could check too. I don't think we'll be able to do much without the GUIDs (in the first version) so failing in assert mode doesn't sound too bad to me. It's up to you to decide.

Reply:
OK.  Let's go with an `assert` for now.  I can make the assertion message helpful at least.

Reply:
It's possible some large binaries may not have this flag enabled for all modules but lets see what happens and if necessary relax this in the future.